test/test_traverse.py

```diff
@@ -70,3 +70,56 @@ def test_with_included_source_and_class_filter(self):
         ]
         self.assertEqual(filtered, ["Strong", "Strong"])
 
+    def test_with_depth_limit(self):
+        doc = Document("a **b** c **d**")
+
+        # Zero depth with root not included yields no nodes.
+        filtered = [t.node.__class__.__name__ for t in traverse(doc, depth=0)]
+        self.assertEqual(filtered, [])
+
+        # Zero depth with root included yields the root node.
+        filtered = [
+            t.node.__class__.__name__
+            for t in traverse(doc, depth=0, include_source=True)
+        ]
+        self.assertEqual(filtered, ["Document"])
+
+        # Depth=1 correctly returns the single node at that level.
+        filtered = [t.node.__class__.__name__ for t in traverse(doc, depth=1)]
+        self.assertEqual(filtered, ["Paragraph"])
+
+        # Depth=2 returns the correct node.
```

Comment:
I think it's exactly as I have wished, probably just a small typo here?

```suggestion
        # Depth=2 returns the correct nodes.
```